shared/signup/username/index.tsx

```diff
@@ -2,57 +2,82 @@ import * as React from 'react'
 import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
 import {maxUsernameLength} from '../../constants/signup'
-import {SignupScreen} from '../common'
+import {SignupScreen, errorBanner} from '../common'
 
 type Props = {
+  error: string
+  initialUsername?: string
   onBack: () => void
-  onChangeUsername: (arg0: string) => void
-  onContinue: () => void
-  onLogin: () => void
+  onContinue: (username: string) => void
+  onLogin: (username: string) => void
   usernameTaken: string | null
+  waiting: boolean
 }
 
-const EnterUsername = (props: Props) => (
-  <SignupScreen
-    banners={
-      props.usernameTaken
-        ? [
-            <Kb.Banner
-              key="usernameTaken"
-              actions={[{onClick: props.onLogin, title: `log in as ${props.usernameTaken}?`}]}
-              color="blue"
-              text="Sorry, this username is already taken. Did you mean to"
-            />,
-          ]
-        : null
-    }
-    buttons={[{label: 'Continue', onClick: props.onContinue, type: 'Success'}]}
-    onBack={props.onBack}
-    title={Styles.isMobile ? 'Create account' : 'Create an account'}
-  >
-    <Kb.Box2
-      alignItems="center"
-      gap={Styles.isMobile ? 'small' : 'medium'}
-      direction="vertical"
-      style={styles.body}
-      fullWidth={true}
+const EnterUsername = (props: Props) => {
+  const [username, onChangeUsername] = React.useState(props.initialUsername || '')
+  const disabled = !username || username === props.usernameTaken
+  const onContinue = () => (disabled ? {} : props.onContinue(username))
+  return (
+    <SignupScreen
+      banners={[
+        ...(props.usernameTaken
+          ? [
+              <Kb.Banner
+                key="usernameTaken"
+                actions={[
+                  {
+                    onClick: () => props.onLogin(props.usernameTaken),
+                    title: `log in as ${props.usernameTaken}?`,
+                  },
+                ]}
+                color="blue"
+                text="Sorry, this username is already taken. Did you mean to"
```

Comment:
i know this is from before but it seems truncated or something

Reply:
it's because of how `Banner`'s `actions` prop works works, it shows up as "Sorry ... Did you mean to __log in as username?__". i'll add a comment